packages/richtext-lexical/src/lexical/plugins/TextPlugin/index.tsx

```diff
@@ -0,0 +1,37 @@
+import type { TextFormatType } from 'lexical'
+
+import { useLexicalComposerContext } from '@lexical/react/LexicalComposerContext'
+import { TEXT_TYPE_TO_FORMAT, TextNode } from 'lexical'
+import { type SanitizedClientFeatures } from 'packages/richtext-lexical/src/features/typesClient.js'
+import { useEffect } from 'react'
+
+export function TextPlugin({ features }: { features: SanitizedClientFeatures }) {
+  const [editor] = useLexicalComposerContext()
+
+  useEffect(() => {
+    const disabledFormats = getDisabledFormats(features.enabledFormats as TextFormatType[])
+    if (disabledFormats.length === 0) {
+      return
+    }
+    return editor.registerNodeTransform(TextNode, (textNode) => {
+      disabledFormats.forEach((disabledFormat) => {
+        if (textNode.hasFormat(disabledFormat)) {
+          textNode.toggleFormat(disabledFormat)
+        }
+      })
+    })
```

Comment:
Now that I think about it, I think overriding the FORMAT_TEXT_COMMAND and PASTE_COMMAND (or importDOM) should be more performant in theory than using a transform.

My concern would be that there are other forms of mutation that we might not be considering. For example: browser extensions or Payload/Lexical plugins that have their own commands.

That is, overriding commands might be more performant, but might not cover some unlikely scenarios.

A third option would be to override the TextNode with our own TextNode, and have it override the setFormat or toggleFormat methods.

Maybe I can leave the transform as is now, and if there is a noticeable performance regression we move to the third option.

Thoughts @AlessioGr?